src/backing/client/grpc_client/proto_util_test.cc

```diff
@@ -0,0 +1,208 @@
+    {grpc::StatusCode::OK, StatusCode::kOk},
+    {grpc::StatusCode::CANCELLED, StatusCode::kCancelled},
+    {grpc::StatusCode::UNKNOWN, StatusCode::kUnknown},
+    {grpc::StatusCode::INVALID_ARGUMENT, StatusCode::kInvalidArgument},
+    {grpc::StatusCode::DEADLINE_EXCEEDED, StatusCode::kDeadlineExceeded},
+    {grpc::StatusCode::NOT_FOUND, StatusCode::kNotFound},
+    {grpc::StatusCode::ALREADY_EXISTS, StatusCode::kAlreadyExists},
+    {grpc::StatusCode::PERMISSION_DENIED, StatusCode::kPermissionDenied},
+    {grpc::StatusCode::UNAUTHENTICATED, StatusCode::kUnauthenticated},
+    {grpc::StatusCode::RESOURCE_EXHAUSTED, StatusCode::kResourceExhausted},
+    {grpc::StatusCode::FAILED_PRECONDITION, StatusCode::kFailedPrecondition},
+    {grpc::StatusCode::ABORTED, StatusCode::kAborted},
+    {grpc::StatusCode::OUT_OF_RANGE, StatusCode::kOutOfRange},
+    {grpc::StatusCode::UNIMPLEMENTED, StatusCode::kUnimplemented},
+    {grpc::StatusCode::INTERNAL, StatusCode::kInternal},
+    {grpc::StatusCode::UNAVAILABLE, StatusCode::kUnavailable},
+    {grpc::StatusCode::DATA_LOSS, StatusCode::kDataLoss},
+};
+
+class FromGrpcStatusToStatusTest : public
+    testing::TestWithParam<CorrespondingStatusCode> {
+  // Purposely empty; no fixtures to instantiate.
+};
+
+INSTANTIATE_TEST_SUITE_P(StatusCodeParameters, FromGrpcStatusToStatusTest,
+                         ValuesIn(kStatusCodeMapping));
+
+TEST_P(FromGrpcStatusToStatusTest, ConversionsWork) {
+  auto mapping = GetParam();
+  std::string const message = "test message";
+  auto const original = grpc::Status(mapping.proto, message);
+  auto const expected = Status(mapping.code, message);
+  auto const actual = FromGrpcStatusToStatus(original);
+  EXPECT_EQ(expected, actual);
+}
+
+using ProtoDigestCase = google::cloud::kms::v1::Digest::DigestCase;
+
+// Mapping between `DigestCase` cases and their protobuf equivalents.
+struct CorrespondingDigestCase {
+  DigestCase expected;
+  ProtoDigestCase proto;
+} const kDigestMapping[]{
+    {DigestCase::kSha256, ProtoDigestCase::kSha256},
+    {DigestCase::kSha384, ProtoDigestCase::kSha384},
+    {DigestCase::kSha512, ProtoDigestCase::kSha512},
+};
+
+class FromProtoToDigestCaseTest : public
+    testing::TestWithParam<CorrespondingDigestCase> {
+  // Purposely empty; no fixtures to instantiate.
+};
+
+INSTANTIATE_TEST_SUITE_P(DigestCaseParameters, FromProtoToDigestCaseTest,
+                         ValuesIn(kDigestMapping));
+
+TEST_P(FromProtoToDigestCaseTest, ConversionsWork) {
+  auto mapping = GetParam();
+  EXPECT_EQ(mapping.expected, FromProtoToDigestCase(mapping.proto));
+}
+
+using ProtoCryptoKeyVersion = google::cloud::kms::v1::CryptoKeyVersion;
+
+// Mapping between `CryptoKeyVersionAlgorithm` cases and their protobuf
+// equivalents.
+struct CorrespondingCryptoKeyVersionAlgorithm {
+  CryptoKeyVersionAlgorithm expected;
+  google::cloud::kms::v1::CryptoKeyVersion_CryptoKeyVersionAlgorithm proto;
+} kCryptoKeyVersionAlgorithmMapping[]{
```

Comment:
Nit: separate type definition from variable declaration as discussed previously

Reply:
Handling in #77.